### PR TITLE
DAN-93 Standardize Connectors

### PR DESCRIPTION
**Pull request by yuhongsun96**

### CODE REVIEW

File: backend/danswer/connectors/factory.py [hunk 1]
```diff
@@ -24,45 +25,52 @@ def build_connector(
...
-        if isinstance(connector, RangePullLoader)
-        else connector
-    )
+    return connector
 
 
-def _range_pull_to_pull(range_pull_connector: RangePullLoader) -> PullLoader:
-    class _Connector(PullLoader):
+# TODO this is some jank, rework at some point
+def _pull_to_load_connector(range_pull_connector: PollConnector) -> LoadConnector:
```
Comment (Weves) [inline backend/danswer/connectors/factory.py]: `_pull_to_load_connector` should be `_poll_to_load_connector`?

File: backend/danswer/server/admin.py [hunk 2]
```diff
@@ -81,10 +81,10 @@ def index(
     _: User = Depends(current_admin_user),
 ) -> None:
     # validate that the connector specified by the source / input_type combination
-    # exists AND that the connector_specific_config is valid for that connector type
+    # exists AND that the connector_specific_config is valid for that connector type, should be load
     build_connector(
         source=source,
-        input_type=index_attempt_request.input_type,
+        input_type=index_attempt_request.input_type,  # TODO chris this is wrong for most requests
```
Comment (Weves) [inline backend/danswer/server/admin.py]: Is this TODO supposed to be here?

File: backend/danswer/server/models.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ class UserRoleResponse(BaseModel):
 
 class SearchDoc(BaseModel):
     semantic_identifier: str
-    link: str
+    link: str | None
```
Comment (Weves) [inline backend/danswer/server/models.py:35]: Why would a link be `None`?

**Review by Weves — APPROVED** [review]

File: backend/danswer/server/models.py (hunk 3 above)
Comment (yuhongsun96) [reply backend/danswer/server/models.py:35]: The logic is just propagated from the indexing flow. We wrote it with the assumption that sometimes we may not be able to get a link for all sources, which... may still be true, though we haven't seen any connectors like that so far

File: backend/danswer/server/admin.py (hunk 2 above)
Comment (yuhongsun96) [reply backend/danswer/server/admin.py]: nope!

File: backend/danswer/connectors/factory.py (hunk 1 above)
Comment (yuhongsun96) [reply backend/danswer/connectors/factory.py]: done

**Pull request merged**